vulnerabilities/importers/apache_httpd.py

```diff
@@ -0,0 +1,106 @@
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+from dataclasses import dataclass
+import xml.etree.ElementTree as ET
+
+import requests
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import DataSource
+from vulnerabilities.data_source import DataSourceConfiguration
+
+
+@dataclass
+class ApacheHTTPDDataSourceConfiguration(DataSourceConfiguration):
+    etags: dict
+
+
+class ApacheHTTPDDataSource(DataSource):
+
+    CONFIG_CLASS = ApacheHTTPDDataSourceConfiguration
+    url = "https://httpd.apache.org/security/vulnerabilities-httpd.xml"
+
+    def updated_advisories(self):
+        # Etags are like hashes of web responses. We maintain
+        # (url, etag) mappings in the DB. `create_etag`  creates
+        # (url, etag) pair. If a (url, etag) already exists then the code
+        # skips processing the response further to avoid duplicate work
+
+        if self.create_etag(self.url):
+            data = fetch_xml(self.url)
+            advisories = to_advisories(data)
+            return self.batch_advisories(advisories)
+
+        return []
+
+    def create_etag(self, url):
+        etag = requests.head(url).headers.get("ETag")
+        if not etag:
+            return True
+
+        elif url in self.config.etags:
+            if self.config.etags[url] == etag:
+                return False
+
+        self.config.etags[url] = etag
+        return True
+
+
+def to_advisories(data):
+    advisories = []
+    for issue in data:
+        resolved_packages = []
+        impacted_packages = []
+        for info in issue:
+            if info.tag == "cve":
+                cve = info.attrib["name"]
+
+            if info.tag == "title":
+                summary = info.text
+
+            if info.tag == "fixed":
+                resolved_packages.append(
+                    PackageURL(name="httpd", version=info.attrib["version"], type="generic")
```

Comment:
that's a nit but I prefer putting args in their original order, e.g. `type` first.

Reply:
Also we may want to have something more distinctive:
- what about a namespace or `apache` or `apache.org`?
- or/and a qualifier for `download_url`?
- or if there are more than just httpd, we could create an `apache` type for all Apache Foundation projects? after all they (all Apache projects) share many characteristics that would make them Package URL-type worthy 
See https://svn.apache.org/repos/asf/comdev/projects.apache.org/trunk/data/projects.xml for instance

Reply:
Adding the namespaces, and qualifiers seems fine.

About creating `apache` type, that I don't think is a good idea because that would end up creating ambiguity in case an apache package is packaged, eg `apache-httpd.rpm`. What would then be package type of  `apache-httpd.rpm` ?

Reply:
>What would then be package type of apache-httpd.rpm ?

That's always an `rpm` type then.
The rationale is the format of packaging and metadata determines the type

Reply:
Using `type=apache name=httpd, version=x`